cmd/juju/cloud/addcredential.go

```diff
@@ -166,9 +194,19 @@ func (c *addCredentialCommand) Run(ctxt *cmd.Context) error {
 	}
 	// Check that the supplied cloud is valid.
 	var err error
-	if c.cloud, err = common.CloudOrProvider(c.CloudName, c.cloudByNameFunc); err != nil {
-		if !errors.IsNotFound(err) {
-			return err
+	if !c.Local {
+		if err := c.maybeRemoteCloud(ctxt); err != nil {
```

Comment:
not sure if we need to return error if no remote cloud found when c.Local == false

Reply:
Good point...
The value of c.Local does not matter. However, if we could not find the remote cloud, we should not try to upload but guide the user instead.